src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -124,20 +126,47 @@ public async Task DeployCdkProject(OrchestratorSession session, CloudApplication
             _interactiveService.LogSectionStart("Deploying AWS CDK project",
                 "Use the CDK project to create or update the AWS CloudFormation stack and deploy the project to the AWS resources in the stack.");
 
-            var deploymentStartDate = DateTime.Now;
             // Handover to CDK command line tool
             // Use a CDK Context parameter to specify the settings file that has been serialized.
-            var cdkDeploy = await _commandLineWrapper.TryRunWithResult($"npx cdk deploy --require-approval never -c {Constants.CloudFormationIdentifier.SETTINGS_PATH_CDK_CONTEXT_PARAMETER}=\"{appSettingsFilePath}\"",
+            var cdkDeployTask = _commandLineWrapper.TryRunWithResult( $"npx cdk deploy --require-approval never -c {Constants.CloudFormationIdentifier.SETTINGS_PATH_CDK_CONTEXT_PARAMETER}=\"{appSettingsFilePath}\"",
                 workingDirectory: cdkProjectPath,
                 environmentVariables: environmentVariables,
                 needAwsCredentials: true,
                 redirectIO: true,
                 streamOutputToInteractiveService: true);
 
-            await CheckCdkDeploymentFailure(cloudApplication, deploymentStartDate);
+            var cancellationTokenSource = new CancellationTokenSource();
+            var retrieveStackIdTask = RetreiveStackId(cloudApplication, cancellationTokenSource.Token);
 
+            var deploymentStartDate = DateTime.UtcNow;
+            var firstCompletedTask = await Task.WhenAny(cdkDeployTask, retrieveStackIdTask);
+            // Deployment end date is captured at this point after 1 of the 2 running tasks yields.
+            var deploymentEndDate = DateTime.UtcNow;
+
+            TryRunResult? cdkDeploy = null;
+            if (firstCompletedTask == retrieveStackIdTask)
+            {
+                // If retrieveStackIdTask completes first, that means a stack was created and exists in CloudFormation.
+                // We can proceed with checking for deployment failures.
+
+                var stackId = retrieveStackIdTask.Result;
```

Comment:
If the `RetreiveStackId` method throws an unhandled exception the `.Result` call here will throw an exception which I think will skip all of the error error checking and logging here. For example I assume we still want to get the CDK exit code. In particular if we trigger a CloudFormation throttling error or a transient error on the SDK calls I don't want this to trigger a deployment failure.